lab-doug/test/matchscore-router-test.js

```diff
@@ -0,0 +1,137 @@
+'use strict';
+const expect = require('chai').expect;
+const request = require('superagent');
+const server = require('../server');
+const storage = require('../lib/storage');
+const Matchscore = require('../model/matchscore');
+const port = process.env.PORT || 3000;
+const baseUrl = `localhost:${port}/api/matchscore`;
+
+
+
+describe('Testing matchscore-router module, ', function(){
+  before((done) => {
+    if(!server.isRunning){
+      server.listen(port, () => {
+        server.isRunning = true;
+        console.log('server is running on port: ', port);
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  after((done) => {
+    if(server.isRunning){
+      server.close(() => {
+        console.log('server has been shutdown');
+        done();
+      });
+      return;
+    }
+    done();
+  });
+
+  describe('Testing POST on /api/matchscore endpoint', function(){
+    after((done) => {
+      storage.pool = {};
+      done();
+    });
+    it('should return a matchscore object', function(done){
+      request.post(baseUrl)
+      .send({distance:300,score:569, xCount: 27})
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        expect(res.body.distance).to.equal(300);
+        expect(res.body.score).to.equal(569);
+        expect(res.body.xCount).to.equal(27);
+        done();
+      });
+    });
+  });
+
+  describe('Testing GET on api/matchscore/:uuid', function(){
+    before((done) => {
+      this.tempMatchscore = new Matchscore(400, 555, 10);
```

Comment:
I *really* like that you're storing what you post and only referring to it in your expects with the variable name. This is a super professional way to approach testing.